evap/staff/forms.py

```diff
@@ -116,6 +131,13 @@ def clean(self):
         if self.cleaned_data.get('DELETE') and not self.instance.can_manager_delete:
             raise SuspiciousOperation("Deleting course type not allowed")
 
+    def save(self, *args, **kwargs):
+        course_type = super().save(*args, **kwargs)
+        if "name_en" in self.changed_data or "name_de" in self.changed_data:
+            for course in course_type.courses.filter(state="published"):
+                update_template_cache(course)
```

Comment:
i noticed we have this for-loop in all three `save`-methods, only to call the same method inside each of these loops. i would suggest to move the loop into that method. like this:
```
if "name_en" in ... <same as before>:
    update_template_cache(course_type.courses.filter(state="published"))
```

and then in `update_template_cache`
```
def update_template_cache(courses):
    for course in courses:
        <do what you did previously in update_template_cache>
```

and then, since `warm_up_template_cache` can already be called with multiple courses, we could call it with all the courses at once instead of with each individual one.